sdlib/d/gc/arena.d

```diff
@@ -751,3 +814,104 @@ unittest resizeLarge {
 	// Now full again:
 	assert(pd9.extent.hpd.full);
 }
+
+unittest resizeLargeGrow {
+	import d.gc.arena;
+	shared Arena arena;
+
+	auto base = &arena.base;
+	scope(exit) arena.base.clear();
+
+	import d.gc.emap;
+	static shared ExtentMap emap;
+	emap.tree.base = base;
+
+	import d.gc.region;
+	shared RegionAllocator regionAllocator;
+	regionAllocator.base = base;
+
+	arena.regionAllocator = &regionAllocator;
+
+	PageDescriptor makeAllocLarge(uint pages) {
+		auto ptr = arena.allocLarge(&emap, pages * PageSize);
+		assert(ptr !is null);
+		auto pd = emap.lookup(ptr);
+		assert(pd.extent.address is ptr);
+		assert(pd.extent.size == pages * PageSize);
+		return pd;
+	}
+
+	void checkGrowLarge(PageDescriptor pd, uint pages) {
+		auto e = pd.extent;
+
+		assert(arena.resizeLarge(&emap, e, pages * PageSize));
+		assert(e.size == pages * PageSize);
+
+		// Confirm that the page after the end of the extent is not included in the map:
+		auto pdAfter = emap.lookup(e.address + e.size);
+		assert(pdAfter.extent !is e);
+
+		// Confirm that the extent correctly grew and remapped:
+		for (auto p = e.address; p < e.address + e.size; p += PageSize) {
+			auto probe = emap.lookup(p);
+			assert(probe.extent == e);
+			assert(probe.data == pd.data);
+			pd = pd.next();
+		}
```

Comment:
This time you need to check that what's after the mapping is either another extent or empty.

Reply:
Why wouldn't you do this in there so it's done consistently?

Reply:
It's not like the check is specific to one allocation, we expect this to be true consistently.

Reply:
Ho, apologies. I was confused because it happens before. Sorry it was pretty late when I commented.